hook/iptables.go

```diff
@@ -0,0 +1,209 @@
+
+			return nil
+		}
+
+		logger.GetInstance().Debug("Inserting iptables on host namespace")
+
+		// Get docker interface
+		containerInterface := viper.GetString("iptables.container_bridge_interface")
+
+		// Get task container ports
+		portMappings := info.TaskInfo.GetContainer().GetDocker().GetPortMappings()
+
+		// Get container ip
+		containerIPs, err := c.ContainerGetIPs(info.ContainerID)
+		if err != nil {
+			return err
+		}
+
+		driver, err := iptables.New()
+		if err != nil {
+			return err
+		}
+
+		// Iterate over all container IPs, and for each IP, iterate on container/host binded ports.
+		// Insert needed iptables for each IP and port.
+		for _, containerIP := range containerIPs {
+			// Insert rule for masquerading network flow going out of container. This rule only needs the
+			// container IP
+			masqueradeRule := fmt.Sprintf(
+				iptableHookMasqueradeRuleTemplate,
+				containerInterface,
+				containerIP.String(),
+			)
+			err = driver.Append("nat", "POSTROUTING", strings.Split(masqueradeRule, " ")...)
+			if err != nil {
+				return err
+			}
+
+			for _, port := range portMappings {
+				// Insert rule for translating incoming data on host port to container
+				dnatDestination := []string{containerIP.String(), ":", strconv.Itoa(int(port.GetContainerPort()))}
+				dnatRule := fmt.Sprintf(
+					iptableHookDnatRuleTemplate,
+					containerInterface,
+					port.GetProtocol(),
+					strconv.Itoa(int(port.GetHostPort())),
+					strings.Join(dnatDestination, ""),
+				)
+				err = driver.Append("nat", "PREROUTING", strings.Split(dnatRule, " ")...)
+				if err != nil {
+					return err
+				}
+
+				// Insert rule for masquerading container -> container network flow
+				selfMasqueradeRule := fmt.Sprintf(
+					iptableHookSelfMasqueradeRuleTemplate,
+					containerIP.String(),
+					port.GetProtocol(),
+					containerIP.String(),
+					strconv.Itoa(int(port.GetContainerPort())),
+				)
+				err = driver.Append("nat", "POSTROUTING", strings.Split(selfMasqueradeRule, " ")...)
+				if err != nil {
+					return err
+				}
+
+				// Insert rule for forwarding incoming data on host port to container
+				forwardRule := fmt.Sprintf(
+					iptableHookForwardRuleTemplate,
+					containerIP.String(),
+					containerInterface,
+					containerInterface,
+					port.GetProtocol(),
+					strconv.Itoa(int(port.GetContainerPort())),
+				)
+				err = driver.Append("filter", "FORWARD", strings.Split(forwardRule, " ")...)
+				if err != nil {
+					return err
+				}
+			}
+		}
+
+		return nil
+	},
+}
+
+// RemoveIptablesHook removes iptables injected by injectIptables hook.
+var RemoveIptablesHook = Hook{
+	Name:     "removeIptables",
+	Priority: 0,
+	Execute: func(c container.Containerizer, info *types.ContainerTaskInfo) error {
+		// Do not execute the hook if we are not on bridged network
+		if info.TaskInfo.GetContainer().GetDocker().GetNetwork() != mesos.ContainerInfo_DockerInfo_BRIDGE {
```

Comment:
Can you try to mutualise some code with InsterIptablesHook ? look likes a copy/paste with just little update

Reply:
It is done